ote_sdk/ote_sdk/algo_backends/test_helpers/fixtures.py

```diff
@@ -0,0 +1,342 @@
+
+    The fixture MUST be overriden in algo backend's conftest.py file.
+    """
+    raise NotImplementedError(
+        "The fixture ote_templates_root_dir_fx should be overriden in algo backend"
+    )
+
+
+@pytest.fixture
+def ote_test_domain_fx():
+    """
+    The fixture returns an string that will be used as the 'subject' field in the
+    e2e test system dashboard.
+    At the moment it is supposed that the fixture should returns something like
+    'custom-object-detection'.
+
+    The fixture MUST be overriden in algo backend's conftest.py file.
+    """
+    raise NotImplementedError(
+        "The fixture ote_test_domain_fx should be overriden in algo backend"
+    )
+
+
+@pytest.fixture
+def ote_test_scenario_fx():
+    """
+    The fixture returns an string that will be used as the 'scenario' field in the
+    e2e test system dashboard.
+    At the moment it is supposed that the fixture should returns something like
+    'api' or 'integration' or 'reallife'.
+
+    The fixture may be overriden in algo backend's conftest.py file.
+    """
+    return "api"
+
+
+#
+#########################################################################################
+
+# pylint: disable=invalid-name
+
+
+def ROOT_PATH_KEY():
+    """
+    Constant for storying in dict-s with paths the root path
+    that will be used for resolving relative paths.
+    """
+    return "_root_path"
+
+
+@pytest.fixture
+def dataset_definitions_fx(request):
+    """
+    Return dataset definitions read from a YAML file passed as the parameter --dataset-definitions.
+    Note that the dataset definitions should store the following structure:
+    {
+        <dataset_name>: {
+            'annotations_train': <annotation_file_path1>
+            'images_train_dir': <images_folder_path1>
+            'annotations_val': <annotation_file_path2>
+            'images_val_dir': <images_folder_path2>
+            'annotations_test': <annotation_file_path3>
+            'images_test_dir':  <images_folder_path3>
+        }
+    }
+    """
+    path = request.config.getoption("--dataset-definitions")
+    if path is None:
+        logger.warning(
+            f"The command line parameter '--dataset-definitions' is not set"
+            f"whereas it is required for the test {request.node.originalname or request.node.name}"
+            f" -- ALL THE TESTS THAT REQUIRE THIS PARAMETER ARE SKIPPED"
+        )
+        return None
+    with open(path, encoding="utf-8") as f:
+        data = yaml.safe_load(f)
+    data[ROOT_PATH_KEY()] = osp.dirname(path)
+    return data
+
+
+@pytest.fixture(scope="session")
+def template_paths_fx(ote_templates_root_dir_fx):
+    """
+    Return mapping model names to template paths, received from globbing the
+    folder pointed by the fixture ote_templates_root_dir_fx.
+    Note that the function searches files with name `template.yaml`, and for each such file
+    the model name is the name of the parent folder of the file.
```

Comment:
Do you think that it is a good idea to fix this and read model_template_id from YAML file?

Reply:
I think so, parent folder name must not be meaningful and unique

Reply:
Ok, will fix it

Reply:
Hmmm, after some reviewing I found out that instead of "gen3_mobilenetV2_ATSS" we will have model name "Custom_Object_Detection_Gen3_ATSS".
Please, note that this model name is much longer.

Also please note that this model name will appear in each test id. At the moment test id-s are sufficiently long, e.g.
`test[ACTION-training_evaluation,model-gen3_mobilenetV2_ATSS,dataset-bbcd,num_iters-KEEP_CONFIG_FIELD_VALUE,batch-KEEP_CONFIG_FIELD_VALUE,usecase-reallife]`

But after this renaming we will have test id-s even longer:
`test[ACTION-training_evaluation,model-Custom_Object_Detection_Gen3_ATSS,dataset-bbcd,num_iters-KEEP_CONFIG_FIELD_VALUE,batch-KEEP_CONFIG_FIELD_VALUE,usecase-reallife]`

@Ilya-Krylov , @druzhkov-paul  -- Is it OK for you??

Reply:
not too much difference for me

Reply:
Ok.

Reply:
Fixed